src/components/LyricView.tsx

```diff
@@ -59,29 +59,51 @@ const LyricView: React.FC<LyricViewProps> = (props: LyricViewProps) => {
     let key = 0;
 
     const songLyrics = song.lyrics;
-    const verseKeys = Object.keys(songLyrics);
-    const lyricBlocks: string[][] = Object.values(songLyrics);
+    const presentationOrder = song.presentation?.split(" ");
 
-    for (let i = 0; i < verseKeys.length; ++i) {
-      const verseKey = verseKeys[i];
-      const verseLyrics = lyricBlocks[i];
+    // if the song has a presentation order defined, use it;
+    // otherwise, present in written order
+    if (typeof presentationOrder !== "undefined") {
+      presentationOrder.forEach((verseNumber) => {
+        const verseName: string = getVerseText(verseNumber);
+        const verseLyrics: string[] = songLyrics.get(verseNumber) || [""];
 
-      lyrics.push(
-        <Fragment key={key++}>
-          <div className="ion-margin-vertical"></div>
-        </Fragment>
-      );
-      lyrics.push(
-        <h5 key={key++} className="ion-margin-top">
-          {getVerseText(verseKey)}
-        </h5>
-      );
-      verseLyrics.forEach((line: string) => {
-        lyrics.push(
-          <IonText key={key++} className="lyricVerse" color="dark">
-            <p>{line}</p>
-          </IonText>
-        );
+        lyrics.push(buildLyricBlock(verseName, verseLyrics, key++));
+      });
+    } else {
```

Comment:
I am nervous about the code for this case because it looks like we don't actually have any cases today where presentation is not provided in the json. Were you able to test this path locally somehow?

Reply:
If you were able to test this path locally then I think that's good enough and we don't need a test. This will be hard to test since we don't have any cases for it today.

Reply:
Yeah, locally I set a breakpoint and changed presentation to undefined . It seemed like it worked whenever I did that.